src/symbol.c

```diff
@@ -15,31 +15,24 @@
 
 #ifndef MRB_NO_PRESYM
 
-# undef MRB_PRESYM_MAX
-# define MRB_PRESYM_NAMED(lit, num, type, name) {lit, sizeof(lit)-1},
-# define MRB_PRESYM_UNNAMED(lit, num) {lit, sizeof(lit)-1},
-
-static const struct {
-  const char *name;
-  uint16_t len;
-} presym_table[] = {
 #ifndef MRB_PRESYM_SCANNING
-# include <mruby/presym.inc>
+/* const uint16_t presym_length_table[]   */
+/* const char * const presym_name_table[] */
+# include <mruby/presym/table.h>
```

Comment:
I like the idea. How about redefining `MRB_PRESYM_NAMED` and `MRB_PRESYM_UNNAMED` and include `presym.inc` twice? E.g.

```c
const char * const presym_name_table[] = {
#undef MRB_PRESYM_MAX
#define MRB_PRESYM_NAMED(lit, num, type, name) lit,
#define MRB_PRESYM_UNNAMED(lit, num) lit,
#include <mruby/presym.inc>
};
const uint16_t presym_length_table[] = {
#undef MRB_PRESYM_MAX
#define MRB_PRESYM_NAMED(lit, num, type, name) sizeof(lit)-1,
#define MRB_PRESYM_UNNAMED(lit, num) sizeof(lit)-1,
#include <mruby/presym.inc>
};
```
Pros
 * Changes are minimal and clear
 * Less complex build process

Cons

 *  Trickier C code
 *  `mruby/presym/id.h` and `mruby/presym/table.h` might be more readable.